x/evm/keeper/bank_extension.go

```diff
@@ -0,0 +1,155 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	auth "github.com/cosmos/cosmos-sdk/x/auth/types"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+
+	"github.com/NibiruChain/nibiru/v2/eth"
+	"github.com/NibiruChain/nibiru/v2/x/evm"
+	"github.com/NibiruChain/nibiru/v2/x/evm/statedb"
+)
+
+var (
+	_ bankkeeper.Keeper     = &NibiruBankKeeper{}
+	_ bankkeeper.SendKeeper = &NibiruBankKeeper{}
+)
+
+type NibiruBankKeeper struct {
+	bankkeeper.BaseKeeper
+	StateDB *statedb.StateDB
+}
+
+func (evmKeeper *Keeper) NewStateDB(
+	ctx sdk.Context, txConfig statedb.TxConfig,
+) *statedb.StateDB {
+	stateDB := statedb.New(ctx, evmKeeper, txConfig)
+	evmKeeper.Bank.StateDB = stateDB
+	return stateDB
+}
+
+func (bk NibiruBankKeeper) MintCoins(
+	ctx sdk.Context,
+	moduleName string,
+	coins sdk.Coins,
+) error {
+	// Use the embedded function from [bankkeeper.Keeper]
+	if err := bk.BaseKeeper.MintCoins(ctx, moduleName, coins); err != nil {
+		return err
+	}
+	if findEtherBalanceChangeFromCoins(coins) {
+		moduleBech32Addr := auth.NewModuleAddress(evm.ModuleName)
+		bk.SyncStateDBWithAccount(ctx, moduleBech32Addr)
```

Comment:
We have to follow the original Bank Keeper's interface exactly, or the override won't work as intended

Reply:
In that case, I think we should use `auth.NewModuleAddress(moduleName)` instead of `auth.NewModuleAddress(evm.ModuleName)` because the `MintCoins()` caller can pass in any module name, and it would be incorrect to always sync the balance changes to the `evm.ModuleName`.

Reply:
You just pass the exact argument to the base keeper. No need to use auth